src/Lecture1.hs

```diff
@@ -104,7 +121,16 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+mySum :: Int -> [Int] -> Int
+mySum result myList = 
+    if null myList
+    then result
+    else result + (head myList) + mySum result (tail myList)
+
+strSum :: [Char] -> Int
+strSum str = 
+    let listInt = map (read::String->Int) (words str)
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
    let listInt = map read (words str)
```

So you can even write this function as a one-liner 🙂